Handle ASCII-encoded segmentation UPIDs correctly

blahspam opened the pull request.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Comcast/scte35-go?pullRequest=32) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Comcast/scte35-go?pullRequest=32) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Comcast/scte35-go?pullRequest=32) it.</sub>

blahspam commented:
@sbowman Can you do a quick review of my changes... and sign the CLA so I can eventually merge this?

I've modified your approach and changed it such that the models always use UTF8 but converts from Latin1 on decode and back on encode. This resolves a number of issues that your original fix didn't address around XML/JSON representation and lossless decoding and encoding.

Thanks.

sbowman commented on pkg/scte35/segmentation_upid.go:
```diff
@@ -168,18 +170,15 @@ func (upid *SegmentationUPID) Name() string {
-			stringVal = "."
+	rs := make([]byte, 0, len(b))
+	for i := range b {
+		if b[i] <= 127 {
```
I think this is supposed to be ASCII printable characters, which means the line should be something like:

    if b[i] > 31 && b[i] < 127

sbowman commented on pkg/scte35/segmentation_upid.go:
```diff
@@ -168,18 +170,15 @@ func (upid *SegmentationUPID) Name() string {
+			continue
 		}
-		stringsValues = append(stringsValues, stringVal)
+		rs[i] = byte(46) // "."
```
You can simplify this:

    rs[i] = '.'

sbowman reviewed (commented):
I think the ASCII is supposed to be printable characters, based on page 69 of SCTE_35_2022b.pdf

blahspam commented:
Made changes per your feedback. If theres nothing else I'll merge it in tomorrow.
